packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -181,11 +176,47 @@ bool ArbCore::deliverMessages(
     std::vector<std::vector<unsigned char>> sequencer_batch_items,
     std::vector<std::vector<unsigned char>> delayed_messages,
     const std::optional<uint256_t>& reorg_batch_items) {
-    auto status = message_data_status.load();
-    if (status != MESSAGES_EMPTY) {
-        std::cerr << "unable to deliver messages, status: " << status
-                  << std::endl;
-        return false;
+    uint32_t error_count = 0;
+    auto status_before = MESSAGES_ERROR;
+    while (true) {
+        status_before = message_data_status.load();
+        if (status_before != MESSAGES_EMPTY) {
+            std::cerr << "unable to deliver messages, status: " << status_before
+                      << ", error count: " << error_count << std::endl;
+        } else {
+            auto success = message_data_status.compare_exchange_weak(
+                status_before, MESSAGES_LOADING);
```

Comment:
Let's use `compare_exchange_strong` here. This isn't performance critical in the larger scope of things, and I wouldn't want to spuriously increment `error_count`.

Reply:
:+1: